File: backend/app/schemas/analytics.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict

from app.models.enums import AnalyticsRange, DisturbanceRisk


class AnalyticsRecordBase(BaseModel):
    recorded_at: date
    carbon_captured_tco2e: float
    biodiversity_score: float
    species_count: int
    vegetation_cover_percent: float
    soil_moisture_percent: float
    ecosystem_health_score: float
    disturbance_risk: DisturbanceRisk


class AnalyticsRecordCreate(AnalyticsRecordBase):
    pass


class AnalyticsRecordRead(AnalyticsRecordBase):
    model_config = ConfigDict(from_attributes=True)


class AnalyticsListResponse(BaseModel):
    site_id: uuid.UUID
    range: AnalyticsRange
    records: list[AnalyticsRecordRead]
    latest: AnalyticsRecordRead | None

File: frontend/src/types/analytics.ts
import type { AnalyticsRange, DisturbanceRisk, ObservationCategory } from './enums'

export interface AnalyticsRecord {
  recorded_at: string
  carbon_captured_tco2e: number
  biodiversity_score: number
  species_count: number
  vegetation_cover_percent: number
  soil_moisture_percent: number
  ecosystem_health_score: number
  disturbance_risk: DisturbanceRisk
}

export interface SiteAnalyticsResponse {
  site_id: string
  range: AnalyticsRange
  records: AnalyticsRecord[]
  latest: AnalyticsRecord | null
}

export type CreateAnalyticsRecordRequest = AnalyticsRecord

export interface SpeciesObservation {
  id: string
  species_name: string
  scientific_name: string | null
  category: ObservationCategory
  observation_count: number
  observed_at: string
  confidence_score: number
}

export type SpeciesByCategory = Partial<Record<ObservationCategory, number>>

export interface SpeciesObservationsResponse {
  items: SpeciesObservation[]
  by_category: SpeciesByCategory
}
